libraries/lib-import-export/Import.cpp

```diff
@@ -629,7 +656,44 @@ bool Importer::Import(
    {
       // Try to open the file with this plugin (probe it)
       wxLogMessage(wxT("Opening with %s"),plugin->GetPluginStringID());
+
+#ifdef __WXMAC__
+      std::shared_ptr<std::promise<std::unique_ptr<ImportFileHandle>>> promise = std::make_shared<std::promise<std::unique_ptr<ImportFileHandle>>>();
+      std::future<std::unique_ptr<ImportFileHandle>> future = promise->get_future();
+      std::thread([promise, plugin, fName, pProj]() mutable {
+         try {
+            auto result = plugin->Open(fName, pProj);
```

Comment:
Did you make sure that all calls are thread-safe?

Reply:
Tried it, unfortunately fopen returns false-positive true and after that when you call plugin->Open UI freezes anyway.
I think this is the only option we have. The most vulnerable scenario is when user cancels importing because thread is working until it finishes it's job and by that time we're out of scope so I've used shared_ptr to take care of promise object to be still available (avoiding potential segfault that way)